### PR TITLE
fix-maxBufSize-less-than-preferredBufSize

### PR DESCRIPTION
**Pull request by harrylaou**

### CODE REVIEW

File: client/src/main/scala/caliban/client/SelectionBuilder.scala [hunk 1]
```diff
@@ -73,7 +74,11 @@ sealed trait SelectionBuilder[-Origin, +A] { self =>
                        readFromString[GraphQLResponse](
                          payload,
                          // allow parsing of large payloads
-                         ReaderConfig.withMaxBufSize(payload.length).withMaxCharBufSize(payload.length)
+                         ReaderConfig
+                           .withMaxBufSize(max(max(payload.length, ReaderConfig.maxBufSize), ReaderConfig.preferredBufSize))
+                           .withMaxCharBufSize(
+                             max(max(payload.length, ReaderConfig.maxCharBufSize), ReaderConfig.preferredCharBufSize)
```
Comment (kyri-petrou) [inline client/src/main/scala/caliban/client/SelectionBuilder.scala]: Should this one just be `max(payload.length, ReaderConfig.maxCharBufSize)`? The `ReaderConfig.preferredCharBufSize` will always be smaller than `ReaderConfig.maxCharBufSize`.

Comment (kyri-petrou) [reply client/src/main/scala/caliban/client/SelectionBuilder.scala]: Same for above

Comment (harrylaou) [reply client/src/main/scala/caliban/client/SelectionBuilder.scala]: Of course. I have updated the PR.

**Review by kyri-petrou — APPROVED** [review]

**Comment by ghostdogpr** [comment]
Thanks!

**Pull request merged**